network/robus/HAL/zephyr/robus_hal.c

```diff
@@ -45,7 +47,82 @@ static void RobusHAL_TimeoutInit(void);
 static void RobusHAL_GPIOInit(void);
 static void RobusHAL_RegisterPTP(void);
 
-/////////////////////////Luos Library Needed function///////////////////////////
+static void RobusHAL_UART_ISR(const struct device *dev, void *user_data)
+{
+    RobusHAL_ResetTimeout(CONFIG_LUOS_TIMEOUT_USEC);
+
+    if(!uart_irq_update(luos_serial))
+    {
+        return;
+    }
+
+    if(uart_irq_tx_ready(luos_serial)) 
+    {
+        uint8_t c;
+
+        if(ring_buf_is_empty(&out_ringbuf)) 
+        {
+            uart_irq_tx_disable(luos_serial);
+            RobusHAL_SetTxState(false);
+        }
+        else 
+        {
+            RobusHAL_ResetTimeout(0);
+            ring_buf_get(&out_ringbuf, &c, 1);
+            uart_fifo_fill(luos_serial, &c, 1);
+            LOG_DBG("Transmitting the following byte from serial 0x%x", c);
+        }
+        k_sem_give(&net_activity_sem);
```

Comment:
If I understand it, you use it to trigger the Luos_loop task.
Here is a few thought about running Luos_loop like that.
We will need a deeper check, but I think:
 - Luos_loop don't need to do anything in case of TX.
 - Luos_loop need to run only one time per complete message (but you can't manage it at HAL level, that is why a macro is interesting here)
 - Luos_loop check time during execution to trigger auto-update messages. We have to find a way to avoid this polling strategy to be able to male it completely event driven.